static/app/views/alerts/rules/issue/index.tsx

```diff
@@ -1204,7 +1204,9 @@ class IssueRuleEditor extends DeprecatedAsyncComponent<Props, State> {
                 disabled={disabled}
                 priority="danger"
                 confirmText={t('Delete Rule')}
-                onConfirm={this.handleDeleteRule}
+                onConfirm={() => {
+                  this.handleDeleteRule();
```

Comment:
Because the behavior changes for promises, I had to find any existing usages where promises were being returned and modify them to no longer do so. Some of these might be better if they waited but I'm just going to keep the previous behavior